File: library/kotlin/src/io/envoyproxy/envoymobile/StatsClientImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.EnvoyEngine

/**
 * Envoy implementation of `StatsClient`.
 */
internal class StatsClientImpl constructor(
  internal val engine: EnvoyEngine
) : StatsClient {

  override fun counter(vararg elements: Element): Counter {
    return CounterImpl(engine, elements.asList())
  }

  override fun gauge(vararg elements: Element): Gauge {
    return GaugeImpl(engine, elements.asList())
  }
}

File: library/swift/src/StatsClientImpl.swift
@_implementationOnly import EnvoyEngine
import Foundation

/// Envoy implementation of StatsClient.
@objcMembers
final class StatsClientImpl: NSObject {
  private let engine: EnvoyEngine

  init(engine: EnvoyEngine) {
    self.engine = engine
    super.init()
  }
}

extension StatsClientImpl: StatsClient {
  func getCounter(elements: [String]) -> Counter {
    return Counter(elements: elements, engine: self.engine)
  }
}
